src/en/flamecomics/src/eu/kanade/tachiyomi/extension/en/flamecomics/FlameComics.kt

```diff
@@ -4,62 +4,158 @@ import android.graphics.Bitmap
 import android.graphics.BitmapFactory
 import android.graphics.Canvas
 import android.graphics.Rect
-import eu.kanade.tachiyomi.multisrc.mangathemesia.MangaThemesia
+import eu.kanade.tachiyomi.network.GET
 import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
 import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.decodeFromString
 import okhttp3.Interceptor
 import okhttp3.MediaType.Companion.toMediaType
 import okhttp3.Protocol
+import okhttp3.Request
 import okhttp3.Response
 import okhttp3.ResponseBody.Companion.toResponseBody
 import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
 import java.io.ByteArrayOutputStream
+import java.net.URLDecoder
 
-class FlameComics : MangaThemesia(
-    "Flame Comics",
-    "https://flamecomics.xyz",
-    "en",
-    mangaUrlDirectory = "/series",
-) {
+class FlameComics : ParsedHttpSource() {
+    override val name = "Flame Comics"
+    private val host = "flamecomics.xyz"
+    override val baseUrl = "https://$host"
+    override val lang = "en"
+    override val supportsLatest = true
+    override val versionId: Int = 2
 
-    // Flame Scans -> Flame Comics
-    override val id = 6350607071566689772
+    private val cdn = "https:cdn.$host"
 
     override val client = super.client.newBuilder()
         .rateLimit(2, 7)
         .addInterceptor(::composedImageIntercept)
         .build()
 
-    override val pageSelector = "div#readerarea img:not(noscript img)[class*=wp-image]"
+    private val removeSpecialCharsregex = Regex("[^A-Za-z0-9 ]")
 
-    // Split Image Fixer Start
-    private val composedSelector: String = "#readerarea div.figure_container div.composed_figure"
+    private fun getRealUrl(string: String?): String {
+        if (string == null) return ""
+        var url = URLDecoder.decode(string.substringAfter("url="), "utf8")
+        url = url.substringBefore("?")
+        url = url.substringBefore("&")
+        return url
+    }
 
-    override fun pageListParse(document: Document): List<Page> {
-        val hasSplitImages = document
-            .select(composedSelector)
-            .firstOrNull() != null
+    override fun chapterFromElement(element: Element): SChapter =
+        throw UnsupportedOperationException()
+
+    override fun chapterListSelector(): String = throw UnsupportedOperationException()
+    override fun imageUrlParse(document: Document): String = ""
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val query = removeSpecialCharsregex.replace(response.request.url.queryParameter("search").toString().lowercase(), "")
```

Comment:
Are you removing special characters since you're not using `HttpUrl` elsewhere, or because the site doesn't support it?

Reply:
I had a problem while Migrating, because the apostrophe in the old title and in the new were different.
And its easier to search, because it will find the manga since it wont matter if you add special chars or not